.github/workflows/tests.yml

```
@@ -24,18 +25,39 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        ruby-version: ["3.0", "3.1", "3.2", "3.3"]
+        ruby-version: ["3.1", "3.2", "3.3", "3.4"]
         rails-version:
-          - "6.1.7.6"
-          - "7.0.8"
-          - "7.1.1"
+          - "7.0"
+          - "7.1"
+          - "7.2"
+          - "8.0"
           - "main"
-        postgres-version: ["12", "13", "14", "15", "16"]
+        postgres-version: ["14", "15", "16", "17"]
         exclude:
-          - ruby-version: "3.2"
-            rails-version: "6.1.7.6"
-          - ruby-version: "3.3"
-            rails-version: "6.1.7.6"
+          - ruby-version: "3.1"
+            postgres-version: "14"
+            rails-version: "main"
+          - ruby-version: "3.1"
+            postgres-version: "15"
+            rails-version: "main"
+          - ruby-version: "3.1"
+            postgres-version: "16"
+            rails-version: "main"
+          - ruby-version: "3.1"
+            postgres-version: "17"
+            rails-version: "main"
+          - ruby-version: "3.1"
+            postgres-version: "14"
+            rails-version: "8.0"
+          - ruby-version: "3.1"
+            postgres-version: "15"
+            rails-version: "8.0"
+          - ruby-version: "3.1"
+            postgres-version: "16"
+            rails-version: "8.0"
+          - ruby-version: "3.1"
+            postgres-version: "17"
+            rails-version: "8.0"
```

Why is `postgres-version` being added to these if we're adding exclusions for all postgres versions for each combination of ruby & rails that's being excluded?

Does it not meaningfully behave the same as this, but with far more checks needing to be run (and being significantly harder to read)
```suggestion
          - ruby-version: "3.1"
            rails-version: "main"
          - ruby-version: "3.1"
            rails-version: "8.0"
```